addons/auth_ldap/models/res_company_ldap.py

```diff
@@ -118,10 +118,18 @@ def _authenticate(self, conf, login, password):
             conn = self._connect(conf)
             conn.simple_bind_s(dn, to_text(password))
             conn.unbind()
-        except ldap.INVALID_CREDENTIALS:
+        except ldap.INVALID_CREDENTIALS as e:
+            _logger.error('An LDAP User Id or password is incorrect: %s', e)
+            return False
+        except ldap.TIMEOUT as e:
+            _logger.error('An LDAP timeout exception occurred: %s', e)
+            return False
+        except ldap.SERVER_DOWN as e:
+            _logger.error('An LDAP server down exception occurred: %s', e)
```

Comment:
none of those changes are necessary. There are many more kinds of exception and we don't need a specific `except` clause for each. Just let them fall through to the `ldap.LDAPError` case, the log message will contain the exact error.